core/workspace_svg.js

```diff
@@ -2724,3 +2597,215 @@ Blockly.WorkspaceSvg.prototype.getAudioManager = function() {
 Blockly.WorkspaceSvg.prototype.getGrid = function() {
   return this.grid_;
 };
+
+
+/**
+ * Registers all of the context menu options for the workspace menu.
+ */
+(function() {
```

Comment:
Now that none of these functions directly reference the workspace (they all use scope), there's no reason to keep them in workspace_svg instead of pulling them out into a separate file. `ws_context_menu_options.js`, maybe? Or else `contextmenu_items` and have it include the block ones as well?

Once it's in a separate file, also break the function apart into helpers (undo, redo, etc).

Reply:
Agreed, moving it to the new file should also help avoid the need for the IIFE (immediately invoking function expression).

Reply:
Done. Plan to add block functions to the same file (contextmenu_items)